packages/react/src/context/slash-id-context.tsx

```diff
@@ -23,12 +23,13 @@ export interface SlashIDProviderProps {
   children: React.ReactNode;
 }
 
-type SDKState =
-  | "initial"
-  | "loaded"
-  | "retrievingToken"
-  | "authenticating"
-  | "ready";
+export enum SDKState {
+  Initial = "initial",
+  Loaded = "loaded",
+  RetrievingToken = "retrievingToken",
+  Authenticating = "authenticating",
+  Ready = "ready"
+}
```

Comment:
Please don't use TS enums 💀 We've had tremendous amount of issues with them in the past, es they do not have proper values in the runtime. I don't think it's worth to introduce them here

Reply:
I'm also up for discussion if you think otherwise @jake-slashid @ikovic

Reply:
Can you explain what exactly the problem was, or give me some reading?

I've used enums for as long as I've used ts, and never experienced any problem. At compile time this just becomes an object.

[post transpilation example](https://www.typescriptlang.org/play?#code/KYOwrgtgBAygIgaRgFwIbOFA3gKClASRAEtljUAbKAXigCJiSzK6AaPKAGQHtUATYHxr0KvAXzYcASsGQAnYsABujAOYAVbgGtQwunNkLlazTpCT8AQTDIAFqDIBjdGr2ob9kE5chVFqDL8AJ56BsF0OAC+QA)

Reply:
We had TS errors when trying to use Core SDK in the React SDK, I remember it was connected to the Factors 🤔 You can see how we pretty much removed them from the Core SDK here: https://github.com/slashid/ng-evangelion/pull/610

Reply:
🤔 

Let me try to reproduce something like that

Reply:
So here's the outcome of that investigation, everything seems in order to me! The `*.d.ts` file looks sensible.

In the transpiled js whenever it's not necessary to actually reference, enum refs are converted to their literal value:
```ts
// before
if (!sid) {
  return;
}

setState(SDKState.Authenticating); // it's an enum

try {
  const identifier =
    factor.method === "oidc" || handle === undefined
      ? null
      : {
          type: handle.type as unknown as PersonHandleType,
          value: handle.value,
        };

  // @ts-expect-error TODO make the identifier optional
  const user = await sid.id(oid, identifier, factor);
  storeUser(user);
  setState(SDKState.Ready);
  return user;
} catch (e) {
  setState(SDKState.Ready);
  logOut();
  throw e;
}

// after
if (!!se) {
  _("authenticating"); // it's a string now
  try {
    const Oe = Z.method === "oidc" || re === void 0 ? null : {
      type: re.type,
      value: re.value
    }, de = await se.id(f, Oe, Z);
    return U(de), _("ready"), de;
  } catch (Oe) {
    throw _("ready"), J(), Oe;
  }
}
```

For cases where there is a reference, that can't be made literal this happens:
```ts
// main.js
const g9 = Object.values(sv).filter((f) => f != sv.Ready);
export {
  g9 as sdkNotReadyStates
}

// main.d.ts
import { SDKState } from "./context/slash-id-context";
export declare const sdkNotReadyStates: (SDKState.Initial | SDKState.Loaded | SDKState.RetrievingToken | SDKState.Authenticating)[];
```

Reply:
I do not have an example handy, the issue we faced was as Kasper described - enum values were not present during runtime. It was a specific set of circumstances though - I believe some of these enums were generated from the API spec, then imported to the core SDK and when referencing them from projects that install the core SDK their values wouldn't be present during runtime.

Is there anything we cannot do with an union specifically to complete this task @jake-slashid ?

Reply:
@ikovic this line of code depends on it, but it's not super important.

https://github.com/slashid/javascript/pull/89/files#diff-3b087a4ccb39999a6946731aae6334d8e3ed07d09a106ffd27fc74763c567ac0R21-R23

To remove the enum, this:
```ts
export const sdkNotReadyStates = Object
  .values(SDKState)
  .filter((state): state is Exclude<SDKState, SDKState.Ready> => state != SDKState.Ready)
```

Becomes this:
```ts
export const sdkNotReadyStates: SDKState[] = [
  "initial",
  "loaded",
  "retrievingToken",
  "authenticating",
  "ready"
]
```

Reply:
The trade-off is that `sdkNotReadyStates` might go stale in future, compiler doesn't help us here.

Reply:
done